rust/ql/lib/codeql/rust/internal/TypeInference.qll

```diff
@@ -540,9 +548,11 @@ private predicate typeEquality(AstNode n1, TypePath prefix1, AstNode n2, TypePat
       let.getInitializer() = n2
     )
     or
-    n1 = n2.(IfExpr).getABranch()
-    or
-    n1 = n2.(MatchExpr).getAnArm().getExpr()
+    n2 =
+      any(MatchExpr me |
+        n1 = me.getAnArm().getExpr() and
+        me.getNumberOfArms() = 1
+      )
```

Comment:
Is the case of an `if` with no `else` analogous to a `match` with only one arm (above), or a function with only one return (below)???

Reply:
`if`s without `else` are hardly interesting, since the type must be `()`.